// format-logs/lib/formatter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.info = exports.warn = exports.err = exports.sub = exports.title = exports.compose = void 0;
const code = (id) => `\x1b[${id}m`;
const format = (id) => (text) => {
    if (text === '')
        return '';
    if (id === undefined)
        return `${text}`;
    return `${code(id)}${text}${BREAKER}`;
};
const BREAKER = code(0);
const fontColors = {
    black: format(30),
    red: format(31),
    green: format(32),
    yellow: format(33),
    blue: format(34),
    magenta: format(35),
    cyan: format(36),
    white: format(37),
    brightBlack: format(90),
    brightRed: format(91),
    brightGreen: format(92),
    brightYellow: format(93),
    brightBlue: format(94),
    brightMagenta: format(95),
    brightCyan: format(96),
    brightWhite: format(97),
};
const bgColors = {
    bgBlack: format(40),
    bgRed: format(41),
    bgGreen: format(42),
    bgYellow: format(43),
    bgBlue: format(44),
    bgMagenta: format(45),
    bgCyan: format(46),
    bgWhite: format(47),
    bgBrightBlack: format(100),
    bgBrightRed: format(101),
    bgBrightGreen: format(102),
    bgBrightYellow: format(103),
    bgBrightBlue: format(104),
    bgBrightMagenta: format(105),
    bgBrightCyan: format(106),
    bgBrightWhite: format(107),
};
const styles = {
    bold: format(1),
    light: format(2),
    italic: format(3),
    underline: format(4),
    crossed: format(9),
};
const base = {
    ...fontColors,
    ...bgColors,
    ...styles,
};
const compose = (text, ...ms) => {
    if (text === '')
        return '';
    return `${ms.reduce((a, b) => base[b](a), text)}`;
};
exports.compose = compose;
const title = (text) => {
    if (text === '')
        return '';
    return `ðŸš€ ${(0, exports.compose)(` ${text} `, 'bgGreen', 'brightWhite', 'bold')}`;
};
exports.title = title;
const sub = (text) => {
    if (text === '')
        return '';
    return `#ï¸âƒ£  ${(0, exports.compose)(text, 'cyan', 'underline')}`;
};
exports.sub = sub;
const err = (text) => {
    if (text === '')
        return '';
    return `ðŸ”´ ${(0, exports.compose)(' ERROR: ', 'bgRed', 'brightWhite', 'bold')} ${(0, exports.compose)(text, 'red')}`;
};
exports.err = err;
const warn = (text) => {
    if (text === '')
        return '';
    return `ðŸŒ¡ï¸  ${(0, exports.compose)(' WARNING: ', 'bgYellow', 'brightWhite', 'bold')} ${(0, exports.compose)(text, 'yellow')}`;
};
exports.warn = warn;
const info = (text) => {
    if (text === '')
        return '';
    return `ðŸ‘€ ${(0, exports.compose)(' INFO: ', 'bgCyan', 'brightWhite', 'bold')} ${(0, exports.compose)(text, 'cyan')}`;
};
exports.info = info;
