// format-logs/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.log = exports.info = exports.warn = exports.err = exports.sub = exports.title = exports.custom = void 0;
var formatter_1 = require("./formatter");
Object.defineProperty(exports, "custom", { enumerable: true, get: function () { return formatter_1.compose; } });
Object.defineProperty(exports, "title", { enumerable: true, get: function () { return formatter_1.title; } });
Object.defineProperty(exports, "sub", { enumerable: true, get: function () { return formatter_1.sub; } });
Object.defineProperty(exports, "err", { enumerable: true, get: function () { return formatter_1.err; } });
Object.defineProperty(exports, "warn", { enumerable: true, get: function () { return formatter_1.warn; } });
Object.defineProperty(exports, "info", { enumerable: true, get: function () { return formatter_1.info; } });
var builder_1 = require("./builder");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return builder_1.log; } });
